src/components/LocationDropdown.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { Fragment } from 'react'
+import { Listbox, Transition } from '@headlessui/react'
+import { LOCATIONS } from '../constants/locations';
+import { CheckIcon, ChevronUpDownIcon } from '@heroicons/react/20/solid'
+
+
+export type LocationType = {
+  country: string;
+  value: number;
+}
+
+type LocationDropdownType = {
+  selectedLocation: LocationType;
+  setSelectedLocation: (location: LocationType) => void;
+}
+
+export const LocationDropdown = ({ selectedLocation, setSelectedLocation }: LocationDropdownType) => {
+
+  return (
+    <div className="fixed top-16 w-72">
+      <Listbox value={selectedLocation} onChange={setSelectedLocation}>
+        <div className="relative mt-1">
+          <Listbox.Button className="relative w-full cursor-default rounded-lg bg-white py-2 pl-3 pr-10 text-left shadow-md focus:outline-none focus-visible:border-indigo-500 focus-visible:ring-2 focus-visible:ring-white focus-visible:ring-opacity-75 focus-visible:ring-offset-2 focus-visible:ring-offset-orange-300 sm:text-sm">
+
+            <span className="block truncate">{selectedLocation ? selectedLocation.country : 'Select a location'}</span>
+            <span className="pointer-events-none absolute inset-y-0 right-0 flex items-center pr-2">
+              <ChevronUpDownIcon
+                className="h-5 w-5 text-gray-400"
+                aria-hidden="true"
+              />
+            </span>
+          </Listbox.Button>
+          <Transition
+            as={Fragment}
+            leave="transition ease-in duration-100"
+            leaveFrom="opacity-100"
+            leaveTo="opacity-0"
+          >
+            <Listbox.Options className="absolute mt-1 max-h-60 w-full overflow-auto rounded-md bg-white py-1 text-base shadow-lg ring-1 ring-black ring-opacity-5 focus:outline-none sm:text-sm">
+              {LOCATIONS.map((location, locationIdx) => (
```

Comment:
yeah i think that would be better :)

Reply:
I think in this case we'd not likely run into problems as the order of the list doesn't change but say if we wanted to change it in the future so that their recent selection is at the top we'd get into trouble

Reply:
That makes sense, good catch!